src/diff/index.js

```diff
@@ -402,6 +402,14 @@ function diffElementNodes(
 				break;
 			}
 		}
+
+		if (
+			dom == null &&
+			excessDomChildren.length == 1 &&
+			excessDomChildren[0] != null
+		) {
+			removeNode(excessDomChildren[0]);
```

Comment:
For Resumed hydration this could be safe as lazy/suspending components return a single VNode, however we are talking about dom-children here, which means that when the VNode returned is a Fragment returning multiple DOM-nodes our heuristic would only match the first DOM-node. Which I guess might also be an issue in how we resume hydration at https://github.com/preactjs/preact/blob/main/src/diff/index.js#L55